// locutus/python/itertools/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.zip_longest = exports.product = exports.permutations = exports.pairwise = exports.islice = exports.compress = exports.combinations_with_replacement = exports.combinations = exports.chain = exports.batched = exports.accumulate = void 0;
var accumulate_ts_1 = require("./accumulate.js");
Object.defineProperty(exports, "accumulate", { enumerable: true, get: function () { return accumulate_ts_1.accumulate; } });
var batched_ts_1 = require("./batched.js");
Object.defineProperty(exports, "batched", { enumerable: true, get: function () { return batched_ts_1.batched; } });
var chain_ts_1 = require("./chain.js");
Object.defineProperty(exports, "chain", { enumerable: true, get: function () { return chain_ts_1.chain; } });
var combinations_ts_1 = require("./combinations.js");
Object.defineProperty(exports, "combinations", { enumerable: true, get: function () { return combinations_ts_1.combinations; } });
var combinations_with_replacement_ts_1 = require("./combinations_with_replacement.js");
Object.defineProperty(exports, "combinations_with_replacement", { enumerable: true, get: function () { return combinations_with_replacement_ts_1.combinations_with_replacement; } });
var compress_ts_1 = require("./compress.js");
Object.defineProperty(exports, "compress", { enumerable: true, get: function () { return compress_ts_1.compress; } });
var islice_ts_1 = require("./islice.js");
Object.defineProperty(exports, "islice", { enumerable: true, get: function () { return islice_ts_1.islice; } });
var pairwise_ts_1 = require("./pairwise.js");
Object.defineProperty(exports, "pairwise", { enumerable: true, get: function () { return pairwise_ts_1.pairwise; } });
var permutations_ts_1 = require("./permutations.js");
Object.defineProperty(exports, "permutations", { enumerable: true, get: function () { return permutations_ts_1.permutations; } });
var product_ts_1 = require("./product.js");
Object.defineProperty(exports, "product", { enumerable: true, get: function () { return product_ts_1.product; } });
var zip_longest_ts_1 = require("./zip_longest.js");
Object.defineProperty(exports, "zip_longest", { enumerable: true, get: function () { return zip_longest_ts_1.zip_longest; } });
